### PR TITLE
ci: implement test cases

### PR DESCRIPTION
**Pull request by l-zeuch**
**Please describe the changes this pull request does and why it should be merged:**
Add automated tests. We've wanted to do this a long time ago, but considering how little there was to test it wasn't really needed until now.

The framework used for tests is [vader.vim](https://github.com/junegunn/vader.vim). We use a `Makefile` to make running tests easier, both locally and via the (rather limited) GitHub Action suite.

The dependency on vader.vim has been declared as dev-dependency in both the README and CONTRIBUTING.

CODEROWNERS got an update to reflect who wrote what file.

**Terms**
- [x] I agree to follow this project's [Code of Conduct](CODE_OF_CONDUCT.md)
- [x] I have read and understood this project's [Contributing Guidelines](../CONTRIBUTING.md)

### CODE REVIEW

**Comment by l-zeuch** [comment]
Okay I'm not too sure how to install our plugin properly here. Happy to receive input.

**Comment by l-zeuch** [comment]
Though this might be out of scope -- since we're already somewhat affecting a lot of files, I'd propose to simultaneously flatten the folder structure, as the `yagpdbcc` subfolders in `ftdetect` et al. are quite redundant.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+test: .bundle/vader.vim
```
Comment (l-zeuch) [inline Makefile:1]: Here I might need to find a way to remove the build artifacts after running `make test` locally, though this isn't of high priority currently.

File: test/clipboard.vader [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Test case file for yanking to system clipboard.
+# vim: set ft=vader
```
Comment (l-zeuch) [inline test/clipboard.vader:2]: Maybe also set the tabwidth and such using the mod line

File: test/syntax.vader [hunk 3]
```diff
@@ -0,0 +1,177 @@
+# Test case file for syntax matching
+# vim: set ft=vader
```
Comment (l-zeuch) [inline test/syntax.vader:2]: see above

**Comment by l-zeuch** [comment]
Should we amend CODEOWNERS according to who wrote which file within this pull request, or leave it as both?

**Comment by LRitzdorf** [comment]
CODEOWNERS updates seem like a good idea, yes

File: test/clipboard.vader (hunk 2 above)
Comment (LRitzdorf) [reply test/clipboard.vader:2]: If you have Vader installed, I think it sets tab width and such automatically, through its ftdetect stage? I'll check that soon

Comment (l-zeuch) [reply test/clipboard.vader:2]: Possibly, though that would require contributors to have vader installed in the first place.
I'll add this as a dependency for development with a README commit, but I believe we should have the modline as failsafe anyway.

Comment (LRitzdorf) [reply test/clipboard.vader:2]: Mm, good point. Especially since Vader complains if you don't get the formatting exactly right

Comment (l-zeuch) [reply test/clipboard.vader]: Done in e87f5e2

File: Makefile (hunk 1 above)
Comment (l-zeuch) [reply Makefile:1]: Resolved in b3fce85 with a new target `clean`

**Comment by l-zeuch** [comment]
I believe this is now ready for review. Let me know if there's still some bits missing.

**Review by LRitzdorf — APPROVED** [review]
Looks good! Will have to add more test cases in the future, but this is a great start. I'll probably open another PR for that soon

**Pull request merged**